extensions/sql-database-projects/src/common/utils.ts

```diff
@@ -156,14 +156,15 @@ export function convertSlashesForSqlProj(filePath: string): string {
 /**
  * Read SQLCMD variables from xmlDoc and return them
  * @param xmlDoc xml doc to read SQLCMD variables from. Format must be the same that sqlproj and publish profiles use
+ * @param publishProfile true if reading from publish profile
  */
-export function readSqlCmdVariables(xmlDoc: any): Record<string, string> {
+export function readSqlCmdVariables(xmlDoc: any, publishProfile: boolean): Record<string, string> {
 	let sqlCmdVariables: Record<string, string> = {};
 	for (let i = 0; i < xmlDoc.documentElement.getElementsByTagName(constants.SqlCmdVariable)?.length; i++) {
 		const sqlCmdVar = xmlDoc.documentElement.getElementsByTagName(constants.SqlCmdVariable)[i];
 		const varName = sqlCmdVar.getAttribute(constants.Include);
 
-		if (sqlCmdVar.getElementsByTagName(constants.DefaultValue)[0] !== undefined) {
+		if (!publishProfile && sqlCmdVar.getElementsByTagName(constants.DefaultValue)[0] !== undefined) {
```

Comment:
Added a comment explaining the difference for Value and DefaultValue for sqlproj and publish profiles. Publish profiles have the actual value to use for a sqlcmd variable, so it doesn't make sense to also have a "default" value since there's only one value it should be using. Sqlproj can have both because in VS the Value is used like `<Value>$(SqlCmdVar__1)</Value>` where the value is the local variable values you can set in VS in the properties. If it isn't set in the properties, the DefaultValue is used, otherwise Value is used. We don't currently have this in ADS, so only DefaultValue is loaded for sqlprojs, then the user can update the value in the publish dialog if they want.